### PR TITLE
Upgrade hapi to v8.x

### PR DESCRIPTION
**Pull request by christophwitzko**
Hi,

this pull request upgrades hapi to v8.x. This might fix issues like #280.

Before merging we need to discuss how much these changes affects the plugins that are using the [static plugin hook](https://github.com/hoodiehq/hoodie-plugin-template/blob/master/hooks/static.js).

best regards,
Christoph

### CODE REVIEW

File: lib/server/plugins/api/index.js [hunk 1]
```diff
@@ -205,11 +204,6 @@ exports.register = function (plugin, options, next) {
       }
     },
     {
-      method: 'HEAD',
```
Comment (christophwitzko) [inline lib/server/plugins/api/index.js]: > HEAD is created automatically for every GET you define.
https://github.com/hapijs/hapi/issues/795#issuecomment-17084121

**Comment by christophwitzko** [comment]
integration tests are failing... :disappointed: investigating...

**Comment by christophwitzko** [comment]
now everything looks fine :+1:

File: lib/server/index.js [hunk 2]
```diff
@@ -8,23 +8,33 @@ module.exports = function () {
...
-      cors: true,
-      payload: {
-        maxBytes: 1048576 * 10 // 10 MB
+      routes: {
+        cors: {
+          override: false
+        },
+        payload: {
+          maxBytes: 1048576 * 10 // 10 MB
+        }
```
Comment (janl) [inline lib/server/index.js]: Maybe consolidate this and the previous connection configuration into a single object, while we are in here.

File: lib/server/plugins/api/index.js (hunk 1 above)
Comment (janl) [inline lib/server/plugins/api/index.js]: :+1:

File: lib/server/index.js (hunk 2 above)
Comment (christophwitzko) [inline lib/server/index.js]: :+1:

**Comment by janl** [comment]
This is very good stuff, thank you! :)

Since this is a major update, I’d like to make sure existing apps still fully work. I’ve done some preliminary tests with a large internal app and that seems fine. Can we find any more folks who are willing to give this a try?

**Comment by christophwitzko** [comment]
@janl thank you so much. :blush:

**Comment by janl** [comment]
Alright, got a few more reports of success, let’s get this merged.

**Comment by christophwitzko** [comment]
:+1: :tada: :relaxed:

**Pull request merged**